streaming/rrdpush.c

```diff
@@ -641,26 +648,41 @@ static int rrdpush_sender_thread_connect_to_master(RRDHOST *host, int default_po
 
     info("STREAM %s [send to %s]: waiting response from remote netdata...", host->hostname, connected_to);
 
+    ssize_t received;
 #ifdef ENABLE_HTTPS
-    if(recv_timeout(&host->ssl,host->rrdpush_sender_socket, http, HTTP_HEADER_SIZE, 0, timeout) == -1) {
+    received = recv_timeout(&host->ssl,host->rrdpush_sender_socket, http, HTTP_HEADER_SIZE, 0, timeout);
+    if(received == -1) {
 #else
-    if(recv_timeout(host->rrdpush_sender_socket, http, HTTP_HEADER_SIZE, 0, timeout) == -1) {
+    received = recv_timeout(host->rrdpush_sender_socket, http, HTTP_HEADER_SIZE, 0, timeout);
+    if(received == -1) {
 #endif
         error("STREAM %s [send to %s]: remote netdata does not respond.", host->hostname, connected_to);
         rrdpush_sender_thread_close_socket(host);
         return 0;
     }
 
-    int answer = strncmp(http, START_STREAMING_PROMPT_V2, strlen(START_STREAMING_PROMPT_V2));
-    if(!answer) {
-        host->labels_flag |= LABEL_FLAG_UPDATE_STREAM;
-        host->labels_flag &= ~LABEL_FLAG_STOP_STREAM;
+    http[received] = '\0';
+    int answer = -1;
+    char *version_start = strchr(http, '=');
+    if(version_start) {
+        version_start++;
+        answer = memcmp(http, START_STREAMING_PROMPT_VN, (size_t)(version_start - http));
+        if(!answer) {
+            rrdpush_set_flags_to_newest_stream(host);
+            host->stream_version = (uint32_t)strtol(version_start, NULL, 10);
```

Comment:
In this case the slave doesn't report the version that was negotiated

Reply:
Hi @stelfrag ,

This part of the code  has relationship with the `master` response. As you can see in the line `664`, I am bringing a termination for the response received, because the unique possibility for we have equals in the answer is the master to work with the version bigger or equal `2`. This is the motive that I start testing the presence of equal before to move for the oldest versions. 
Right now the majority of Netdata installed are the oldest, but in a near future this will invert, so I decided to put the test to new before any other test.

best regards!

Reply:
Yes that's fine but in that case the slave doesn't report with what version it connected. 
It would be nice to have an indication in line 695

```    
info("STREAM %s [send to %s]: established communication - ready to send metrics...", host->hostname, connected_to);
```
That will say with which version it connected. Now with identical versions in master and slave, I get info on the master (for example`Netdata is using the newest stream protocol`) but not in the slave

Reply:
I loved your idea, I brought it with the last commit. :)
Thank you very much!